toolkit/tools/imagegen/diskutils/diskutils.go

```diff
@@ -339,7 +339,34 @@ func CreatePartitions(diskDevPath string, disk configuration.Disk, rootEncryptio
 	// Partitions assumed to be defined in sorted order
 	for idx, partition := range disk.Partitions {
 		partitionNumber := idx + 1
-		partDevPath, err := CreateSinglePartition(diskDevPath, partitionNumber, partitionTableType.String(), partition)
+		partType := "primary"
+
+		// MBR only allows up to four primary partitions, thus check here whether
+		// we need to create extended partition or not in case more than four partitions are specified
+		if partitionNumber >= 4 && len(disk.Partitions) > 4 && partitionTableType == configuration.PartitionTableTypeMbr {
```

Comment:
>4 [](http://example.com/codeflow?start=24&length=1)

Could we use constants for raw numbers? Especially if they are repeated. This way it's clear we always mean the same thing with `4` and it might be easier to understand the intention with something along the lines of `partitionNumber > MBRPartitionLimit`.